src/autocomplete.ts

```diff
@@ -40,6 +42,20 @@ export default class Autocomplete {
       this.feedback.setAttribute('aria-atomic', 'true')
     }
 
+    // if clearButton is not a button, make it one
+    if (this.clearButton && this.clearButton.tagName.toLowerCase() !== 'button') {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const [tagName, ...otherAttributes] = this.clearButton.attributes
+      const newClearButton = document.createElement('button')
+      newClearButton.innerHTML = this.clearButton.innerHTML
+      newClearButton.id = this.clearButton.id
+      for (const attr of otherAttributes) {
+        newClearButton.setAttribute(attr.name, attr.value)
+      }
+      this.clearButton.parentNode?.replaceChild(newClearButton, this.clearButton)
+      this.clearButton = newClearButton
+    }
+
```

Comment:
what is the use case for a user to not set it as a button? I think it can become super confusing to change the user's HTML here.

Reply:
I think of this as "defensive accessibility" where the API ensures you don't end up with an inaccessible element in this component. I've seen the wrong element used enough times that I want to keep this in. Why do you think it'll be confusing?